File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
    public int solution(String myString, String pat) {
        int answer = 0;
        String[] convert = myString.split("");
        
        for(int i=0;i<convert.length; i++){
            if(convert[i].equals("A")) {
            	convert[i] = "B";
            }
            else {
            	convert[i] = "A";
            }  	
        }
        StringBuilder sb = new StringBuilder();
        
        for(String s : convert){
            sb.append(s);
        }
        
        
        if(sb.toString().contains(pat)){
            answer = 1;
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.js
function solution(myString, pat) {
    var answer = 0;
    let newString = '';

    for(let ch of myString){
        if(ch === 'A'){
            newString += 'B';            
        }else{
            newString += 'A';
        }
    }
    
    if(newString.includes(pat)){
        answer = 1
    }
    
    return answer;
}